File: r/best_time.R
box_plots_best_time <- function(){
  
  library(DBI)
  
  conn <- dbConnect(RSQLite::SQLite(), dbname = '/Users/levbarbash/LSE/LondonProg/Coursework/airline2.db')
  
  sql_col <- c('CRSDepTime', 'DayOfWeek', 'DayOfMonth', 'Month')
  plot_col <- c('Hours', 'Day_of_week', 'Day_of_month', 'Month') 
  
  for (i in 1:4) 
  {
    temp_delay <- dbGetQuery(conn, glue::glue('
    SELECT {sql_col[i]} AS `{sql_col[i]}`, DepDelay AS DepDelay
    FROM ontime
    WHERE ontime.Cancelled = 0 AND ontime.Diverted = 0 AND ontime.DepDelay > 0
    '))
    temp_delay[, 1] <- temp_delay[, 1] %/% (1 + 99 * as.integer(sql_col[i] == 'CRSDepTime'))
    colnames(temp_delay)[1] <- 'x_value'
    boxplot(DepDelay~x_value,
            data=temp_delay,
            outline=FALSE,
            xlab=plot_col[i],
            ylab="Delay (in minutes)",
            col="orange",
            border="brown"
    )
  }
  
  dbDisconnect(conn)
}

File: python/best_time.py
def box_plots_best_time():
    
    import sqlite3
    import sql_query
    import pandas as pd
    import numpy as np
    import seaborn as sns
    import matplotlib.pyplot as plt
    
    f, axs = plt.subplots(2, 2, figsize=(12, 8), gridspec_kw=dict(width_ratios=[4, 3]))
    for sql_col, pd_col in zip(['CRSDepTime', 'DayOfWeek', 'DayOfMonth', 'Month'], ['Hours', 'Day of week', 'Day of month', 'Month']):
        temp_delay = sql_query.execute_sql_query('''
        SELECT ''' + sql_col + ''' AS ''' + sql_col + ''', DepDelay as DepDelay
        FROM ontime
        WHERE ontime.Cancelled = 0 AND ontime.Diverted = 0 AND ontime.DepDelay > 0
        ''').rename(columns={0:pd_col, 1:'Delay'})
        temp_delay[pd_col] = temp_delay[pd_col] // (1 + 99 * int(pd_col == 'Hours'))
        temp_axs = axs[int(pd_col == 'Day of month') + int(pd_col == 'Month'), int(pd_col == 'Day of week') + int(pd_col == 'Month')]
        sns.boxplot(data=temp_delay, x=pd_col, y='Delay', showfliers = False, ax=temp_axs)
        temp_axs.set_ylim([0, 160])
    f.tight_layout()